PR for new nginx version 1.18

uhliarik opened the pull request.

hhorak commented:
It looks good to me. Changes from 1.16 only involve version change. All Dockerfiles are disabled though, which is understandable, but maybe Fedora could be enabled and 1.18 added to Makefile -- that would make running the tests for Fedora at least, which is not the case now, without 1.18 being in the Makefile.

uhliarik commented:
> It looks good to me. Changes from 1.16 only involve version change. All Dockerfiles are disabled though, which is understandable, but maybe Fedora could be enabled and 1.18 added to Makefile -- that would make running the tests for Fedora at least, which is not the case now, without 1.18 being in the Makefile.

Thanks Honzo for review, I added 1.18 to Makefile and removed .exclude file for Fedora.

phracek commented:
LGTM from my point of view. Thanks for changing Makefile and removing .exclude-fedora.

phracek commented:
[test][test-openshift]

hhorak commented:
LGTM as well, but tests fail on centos, because `.exclude-centos7` is not there so it tries to run tests for `1.18` on centos7, which fails because `rh-nginx118` does not exist yet. Can you create this `.exclude-centos7` file, please?

hhorak commented:
I don't understand why Fedora test fails:
```
Error: image f31/s2i-core:latest not found
```
`podman pull registry.fedoraproject.org/f31/s2i-core:latest` works for me. Maybe some temporary registry issue?

phracek commented:
I have checked fedora registry as well. I think this could be an infrastructure outage.

phracek commented:
It looks like this test for the newest NginX 1.18 version does not work.

```
test_for_output "http://$(container_ip):${test_port}/" "NGINX is working"
```

default.conf contains localhost2 and not localhost.

See log, it always returns NGINX2 is working.
```
12:23:09 [1m[INFO] Testing URL http://172.17.0.2:8080/ with HTTP hostname localhost produces output NGINX is working...[0m
12:23:09 
12:23:09 172.17.0.1 - - [01/Jul/2020:10:23:09 +0000] "GET / HTTP/1.1" 200 106 "-" "curl/7.29.0" "-"
12:23:09 HTTP/1.1 200 OK
12:23:09 Server: nginx/1.18.0
12:23:09 Date: Wed, 01 Jul 2020 10:23:09 GMT
12:23:09 Content-Type: text/html
12:23:09 Content-Length: 106
12:23:09 Last-Modified: Wed, 01 Jul 2020 10:23:07 GMT
12:23:09 Connection: keep-alive
12:23:09 ETag: "5efc640b-6a"
12:23:09 Accept-Ranges: bytes
12:23:09 
12:23:09 <html>
12:23:09 <head>
12:23:09 	<title>Test NGINX passed</title>
12:23:09 </head>
12:23:09 <body>
12:23:09 <h1>NGINX2 is working</h1>
12:23:09 </body>
12:23:09 </html>
```

My guess, it was changed in 1.16 or 1.18. version 1.16 for Fedora is skipped therefore it was not test it.

uhliarik commented:
[test][test-openshift]

phracek commented:
[test][test-openshift]

phracek commented on 1.18/test/run:
```diff
@@ -0,0 +1,403 @@
+  cat $cid_file
+  info "Testing the HTTP connection (http://$(container_ip):${test_port})"
+
+  if test_for_output "http://$(container_ip):${test_port}/" "NGINX is working" &&
```
```suggestion
  if test_for_output "http://$(container_ip):${test_port}/index.html" "NGINX is working" &&
```

uhliarik commented:
Ughh, I forgot that maintainer in Fedora changed default nginx.conf in 1.18 update, which breaks this testing scenario, since default fallback is not working anymore:

https://src.fedoraproject.org/rpms/nginx/c/58b4d7a8fd727191cb3ad4d8f9803a9d24bd25eb?branch=master

specifically this BZ: 

https://bugzilla.redhat.com/show_bug.cgi?id=1373822

phracek commented:
[test][test-openshift]

phracek commented:
BTW. Thanks for solving this issue. You helped me a lot.

phracek commented:
Let's try test on Fedora

[test]

pkubatrh commented:
lgtm as well, thanks! Tests are green so merging.
